genie-common/src/main/java/com/netflix/genie/common/dto/Command.java

```diff
@@ -107,18 +121,61 @@ protected Command(@Valid final Builder builder) {
          * @param executable The executable for the command
          * @param checkDelay How long the system should go between checking the status of jobs run with this command.
          *                   In milliseconds.
+         * @deprecated Use {@link Command.Builder#Builder(String, String, String, CommandStatus, List, long)}
+         */
+        @Deprecated
+        public Builder(
+            final String name,
+            final String user,
+            final String version,
+            final CommandStatus status,
+            final String executable,
+            final long checkDelay
+        ) {
+            super(name, user, version);
+            this.bStatus = status;
+            this.bExecutable = executable;
+            this.bCheckDelay = checkDelay;
+        }
+
+        /**
+         * Constructor with required fields.
+         *
+         * @param name                   The name to use for the Command
+         * @param user                   The user to use for the Command
+         * @param version                The version to use for the Command
+         * @param status                 The status of the Command
+         * @param executableAndArguments The executable for the command and its fixed arguments
+         * @param checkDelay             How long the system should go between checking the status of jobs run with this
+         *                               command. In milliseconds.
          */
         public Builder(
+            final String name,
+            final String user,
+            final String version,
+            final CommandStatus status,
+            final List<String> executableAndArguments,
+            final long checkDelay
+        ) {
+            super(name, user, version);
+            this.bStatus = status;
+            this.bExecutableAndArguments = ImmutableList.copyOf(executableAndArguments);
+            this.bCheckDelay = checkDelay;
+        }
+
+        /*
+         * This constructor is provided just for JSON deserialization and considers both the old 'executable' and the
+         * new 'executableAndArguments' fields optional for API backward compatibility.
```

Comment:
I don't see any of these constructors marked with `@JsonCreator` so which one is used by Jackson by default?

It seems like the second public on you added really isn't necessary and you can just have this one marked `@JsonCreator` and it will use the `with` methods and then get exceptions on check in the `build`

Reply:
It wasn't marked before either because Jackson is smart enough to figure it out

Reply:
Also, the second constructor i added is the new "proper" constructor with all required arguments.
Once we phase out the old one, we can remove the `withExecutableAndArgument(List)` maybe i should mark it already deprecated